ledger-service/http-json/README.md

```diff
@@ -8,3 +8,67 @@ Documentation can also be found in the RST format:
 - [Search Query Language](/docs/source/json-api/search-query-language.rst)
 - [Metrics of the HTTP JSON API](/docs/source/json-api/metrics.rst)
 
+## Development guidelines
+
+The original endpoints for the JSON API followed some guidelines that
+should be kept in mind when considering extensions and changes to the
+API.
+
+1. The JSON API is here to get basic ACS-using applications off the
+   ground quickly, with a shallow development learning curve.  It is not
+   meant to support every ledger use case or every size of application.
+   Always keep in mind that JSON API is an ordinary ledger client; there
+   is nothing that it can do that any application written against the
+   gRPC API could not do, and in a way that is more optimized for
+   whatever advanced application someone might have in mind.
+
+2. JSON API is not a REST API.  That's one reason why REST isn't in the
+   name, and that we never refer to it as REST anywhere in the
+   documentation.  It uses HTTP, and JSON, not REST.  So "it's what REST
+   does" is not considered at all in API design.
+
+3. Instead, the most influential external design idea is that of
+   minimizing the use of HTTP error codes, in favor of reporting status
+   information in the response body, [as linked from the JSON API
+   documentation](https://blog.restcase.com/rest-api-error-codes-101/).
+   So do not overthink HTTP status; we do report broadly reasonable
+   status, but what matters is that the response error body is
+   semantically useful.
```

Comment:
Should we also mention that we're minimizing the _direct_ use of HTTP codes, despite us using a wide range of those codes as we transparently pass through codes from the Ledger API?

Reply:
c94c23d2bb98606deb192bc3c932b54525726142